### ISSUE TITLE
[Question] How to tie/share weights in a flax neural network

### ISSUE THREAD

**Issue by patrickvonplaten** [google/flax#1263]
Sorry this is not really a bug report but more a question. Lots of language models tie their word embedding matrix to the output logits matrix (This both saves quite some memory since vocab_sizes can be huge & can lead to better results). In PyTorch it's quite easy to do so - one can simply do the following:

```python

SimpleNetwork(nn.Module):

def __init__(self):
       self.input_embedding = nn.Embedding(...)
       self.output_embedding = nn.Dense(...)

model = SimpleNetwork()

model.output_embedding.weight = model.input_embedding.weight
```

This will simply make sure that both weights point to the same weight node in the graph and consequently the weights are tied in the network. 

Is there a way to do this in Flax? 

I tried to simple set the weights of `state` to each other, *e.g.*:

```python
model = SimpleNetwork()

state = model.init(....)

state["params"]["output_embedding"]["kernel"] = state["params"]["input_embedding"]["embedding"].T
```

but this doesn't seem to actually tie the weights (during gradient descent the weights are updated independently). Do you know if there is an elegant way to share weights in Flax?

Thanks a lot!
